File: src/main/java/com/my/admin/app/commerce/service/TurnService.java
package com.my.admin.app.commerce.service;

import com.black.food.manager.dao.TurnDao;
import com.black.food.manager.model.api.response.TurnResponse;
import com.black.food.manager.model.transformer.TurnTransformer;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.Set;
import java.util.stream.Collectors;

@Service
public class TurnService {

    private TurnDao turnDao;

    private TurnTransformer turnTransformer;

    public Set<TurnResponse> findAllTurnResponse(){
        return turnDao.findAll()
                .stream()
                .map(turn -> turnTransformer.apply(turn))
                .collect(Collectors.toSet());
    }

    @Autowired
    public void setTurnDao(final TurnDao turnDao) {
        this.turnDao = turnDao;
    }

    @Autowired
    public void setTurnTransformer(final TurnTransformer turnTransformer) {
        this.turnTransformer = turnTransformer;
    }
}

File: src/main/kotlin/com/bk/reactive/app/commerce/admin/my/service/TurnService.kt
package com.bk.reactive.app.commerce.admin.my.service

import com.bk.reactive.app.commerce.admin.my.dao.TurnDAO
import com.bk.reactive.app.commerce.admin.my.model.api.response.TurnResponse
import com.bk.reactive.app.commerce.admin.my.model.mapper.TurnMapper.Companion.fromTurnToTurnResponse
import org.springframework.stereotype.Component
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

@Component
class TurnService(
    private val turnDAO: TurnDAO
) {
    fun findAllTurnResponse(): Flux<TurnResponse> {
        return turnDAO.findAll()
            .map { fromTurnToTurnResponse(it) }
            .onErrorResume {
                Mono.error { Exception("There are no registered turns") }
            }
    }
}